mne/io/tests/test_apply_function.py

```diff
@@ -40,6 +45,11 @@ def test_apply_function_verbose():
     pytest.raises(TypeError, raw.apply_function, bad_1, n_jobs=2)
     pytest.raises(ValueError, raw.apply_function, bad_2, n_jobs=2)
 
+    # test return type when `channel_wise=False`
+    raw.apply_function(printer, channel_wise=False)
+    pytest.raises(TypeError, raw.apply_function, bad_1, channel_wise=False)
```

Comment:
the "modern" way of writing these is to do:
```
with pytest.raises(TypeError, match='must be blah blah blah'):
    raw.apply_function(bad_1, channel_wise=False)
```
it makes it clearer what's being caught and why, and ensures that's the case